src/intelligence_layer/connectors/studio/studio.py

```diff
@@ -191,3 +206,66 @@ def _upload_trace(self, trace: ExportedSpanList) -> str:
             case _:
                 response.raise_for_status()
         return str(response.json())
+
+    def submit_dataset(
+        self,
+        dataset: Dataset,
+        examples: Iterable[Example[Input, ExpectedOutput]],
+    ) -> str:
+        """Submits a dataset to Studio.
+
+        Args:
+            dataset: :Dataset: to be uploaded
+            examples: :Examples: of Dataset
+
+        Returns:
+            id of created dataset
+        """
+        url = urljoin(self.url, f"/api/projects/{self.project_id}/datasets")
+        source_data_list = [
+            example.model_dump_json()
+            for example in sorted(examples, key=lambda x: x.id)
+        ]
+
+        source_data_file = "\n".join(source_data_list).encode()
+
+        data = {
+            "name": dataset.name,
+            "labels": list(dataset.labels) if dataset.labels is not None else [],
+            "total_datapoints": len(source_data_list),
+        }
+
+        # Handle metadata separately to avoid double JSON encoding
+        if dataset.metadata:
+            if isinstance(dataset.metadata, str):
+                data["metadata"] = dataset.metadata
+            else:
+                data["metadata"] = json.dumps(dataset.metadata)
+
+        response = requests.post(
+            url,
+            files={"source_data": source_data_file},
+            data=data,
+            headers=self._headers,
+        )
+
+        self._raise_for_status(response)
+        return str(response.text)
+
+    def _raise_for_status(self, response: requests.Response) -> None:
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            exception_factory = self._status_code_to_exception.get(
+                HTTPStatus(response.status_code), DataInternalError
+            )
+            raise exception_factory(
+                response.text, HTTPStatus(response.status_code)
+            ) from e
```

Comment:
You are throwing a `DataInternalError` here, which may not be the right error ;). Currently, i believe we don't really need to throw a new error, all we need is to catch the error, print its details and then raise it again